swap/swap.go

```diff
@@ -99,6 +99,29 @@ func New(stateStore state.Store, prvkey *ecdsa.PrivateKey, contract common.Addre
 	return sw
 }
 
+const balancePrefix = "balance_"
```

Comment:
consider using a reverse index where the keys are constructed as:
`peer.String() + suffix` - this will allow you to iterate over all peer-relevant entries in one iteration (because of the shared prefix)

Reply:
The issue I see with this is that it would slow down the querying of all balances for a node, which we need in the `Balances` function. As far as I know, we're not currently iterating over all peer-related entities in the SWAP code, but I'll verify this nonetheless.

Reply:
We are indeed not iterating over peer-related info in the same iteration. All information is accessed at different points in time. Thanks for the suggestion @acud but in this case the opposite seems to make more sense.